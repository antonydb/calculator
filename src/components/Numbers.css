.numbers {
  display: grid;
  grid-auto-rows: 1fr;
  grid-gap: 10px;
  grid-template-columns: repeat(3, 1fr);
}
