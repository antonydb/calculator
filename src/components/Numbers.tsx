import * as React from "react";
import { Button } from "./Button";
import "./Numbers.css";

interface NumbersProps {
  onClick: (val: string) => void;
}

const numbers = Array(10)
  .fill(undefined)
  .map((_, index) => `${index}`);

export const Numbers: React.FC<NumbersProps> = ({ onClick }) => (
  <div className="numbers">
    <Button value="." onClick={onClick}>
      .
    </Button>
    {numbers.map((value) => (
      <Button key={value} value={value} onClick={onClick}>
        {value}
      </Button>
    ))}
  </div>
);
