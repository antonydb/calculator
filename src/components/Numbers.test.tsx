import * as React from "react";
import { render } from "@testing-library/react";
import { Numbers } from "./Numbers";

describe("<Numbers />", () => {
  const { container } = render(<Numbers onClick={() => undefined} />);

  it(`should match snapshot`, () => {
    expect(container.innerHTML).toMatchSnapshot();
  });
});
