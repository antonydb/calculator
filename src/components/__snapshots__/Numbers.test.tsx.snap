// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Numbers /> should match snapshot 1`] = `"<div class=\\"numbers\\"><button class=\\"button button-default\\" type=\\"button\\" value=\\".\\">.</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"0\\">0</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"1\\">1</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"2\\">2</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"3\\">3</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"4\\">4</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"5\\">5</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"6\\">6</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"7\\">7</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"8\\">8</button><button class=\\"button button-default\\" type=\\"button\\" value=\\"9\\">9</button></div>"`;
